gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/support/PubSubConfigurationFactory.java

```diff
@@ -0,0 +1,161 @@
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.gcp.GoogleCloudConfiguration;
+import io.micronaut.gcp.UserAgentHeaderProvider;
+import io.micronaut.gcp.pubsub.configuration.PubSubConfigurationProperties;
+import org.threeten.bp.Duration;
+
+import javax.inject.Singleton;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Factory class to create default settings for PubSub Publisher and subscriber beans.
+ *
+ * @author Vinicius Carvalho
+ * @since 2.0.0
+ *
+ */
+@Factory
+@Requires(classes = Publisher.class)
+public class PubSubConfigurationFactory {
+
+    private final PubSubConfigurationProperties pubSubConfigurationProperties;
+
+    private final GoogleCloudConfiguration googleCloudConfiguration;
+
+    public PubSubConfigurationFactory(PubSubConfigurationProperties pubSubConfigurationProperties, GoogleCloudConfiguration googleCloudConfiguration) {
+        this.pubSubConfigurationProperties = pubSubConfigurationProperties;
+        this.googleCloudConfiguration = googleCloudConfiguration;
+    }
+
+    /**
+     * @return Default retrySettings
+     */
+    @Singleton
+    protected RetrySettings retrySettings() {
+        PubSubConfigurationProperties.Publisher.Retry retryProperties = this.pubSubConfigurationProperties.getPublisher().getRetry();
+        RetrySettings.Builder builder = RetrySettings.newBuilder();
+        RetrySettings settings = ifNotNull(retryProperties.getInitialRetryDelaySeconds(),
+                (x) -> builder.setInitialRetryDelay(Duration.ofSeconds(x)))
+                .apply(ifNotNull(retryProperties.getInitialRpcTimeoutSeconds(),
+                        (x) -> builder.setInitialRpcTimeout(Duration.ofSeconds(x)))
+                .apply(ifNotNull(retryProperties.getJittered(), builder::setJittered)
+                .apply(ifNotNull(retryProperties.getMaxAttempts(), builder::setMaxAttempts)
+                .apply(ifNotNull(retryProperties.getMaxRetryDelaySeconds(),
+                        (x) -> builder.setMaxRetryDelay(Duration.ofSeconds(x)))
+                .apply(ifNotNull(retryProperties.getMaxRpcTimeoutSeconds(),
+                        (x) -> builder.setMaxRpcTimeout(Duration.ofSeconds(x)))
+                .apply(ifNotNull(retryProperties.getRetryDelayMultiplier(), builder::setRetryDelayMultiplier)
+                .apply(ifNotNull(retryProperties.getTotalTimeoutSeconds(),
+                        (x) -> builder.setTotalTimeout(Duration.ofSeconds(x)))
+                .apply(ifNotNull(retryProperties.getRpcTimeoutMultiplier(), builder::setRpcTimeoutMultiplier)
+                .apply(false))))))))) ? builder.build() : null;
+        return settings;
+    }
+
+    /**
+     *
+     * @return batchSettings
+     */
+    @Singleton
+    public BatchingSettings publisherBatchSettings() {
+        BatchingSettings.Builder builder = BatchingSettings.newBuilder();
```

Comment:
Same thing here. Allow Micronaut to bind to `BatchingSettings.Builder` directly using `@ConfigurationBuilder`